src/main/java/io/r2dbc/postgresql/exception/PostgresqlServerErrorException.java

```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package io.r2dbc.postgresql;
+package io.r2dbc.postgresql.exception;
```

Comment:
What's the reason behind moving the exception to a separate package?

Typically, exceptions stay in the same package as the API itself to avoid package cycles and to keep exceptions on the same level as the code that emits the exception.

Reply:
my idea here was to have the exceptions inside the exception related code. just as an organization of packages.

I thought it is confusing if I look into a package and I see stuff like:
- service.class
- exception.class
- factory.class

but, if this is not a problem for this project, I could move the exception back. ( :

Reply:
Separating domain types by technicality is in fact considered an anti-pattern because it distributes things that belong together. Having all domain functionality, that belongs together in a single place keeps code more cohesive and it is easier to identify things that belong together. 

A nice side effect of keeping types in the same package allows hiding internal classes (package private visibility).

Reply:
I got your point. 

I just moved everything out of the package and delete it. 

Before merge, I will squash all commits.

Thank you for the explanation.